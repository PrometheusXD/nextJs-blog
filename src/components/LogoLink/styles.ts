import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  align-items: center;
  color: inherit;

  > img {
    width: 14rem;
    height: 14rem;
    border-radius: 50%;
  }
`;
