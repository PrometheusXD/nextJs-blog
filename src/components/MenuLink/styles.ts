import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display:block;
    padding-bottom: ${theme.spacings.small};
    font-size: 1.8;
    border-right: 0.5rem solid ${theme.colors.primary};
    transition: all 300ms ease-in-out;

    :hover {
      border-right: 0.5rem solid ${theme.colors.secondary};
    }

    > a {
      color:${theme.colors.white};
      text-decoration: none;
      transition: all 300ms ease-in-out;

      :hover {
        color: ${theme.colors.secondary};
      }
    }

  `}
`;
