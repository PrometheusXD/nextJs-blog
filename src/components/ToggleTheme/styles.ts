import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
    position: fixed;
    top: 2rem;
    right: 2rem;
    z-index: 2;
`;

export const Label = styled.label`
    position: relative;
    display: inline-block;
    width: 5rem;
    height: 2.5rem;
    line-height: 0;
    font-size: 0;
    overflow: hidden;
    color: rgba(0,0,0,0);
`;

export const Input = styled.input`
  ${({ theme }) => css`
    appearance: none;
    opacity: 0;
    width: 0;
    height: 0;

    &:checked + ${Slider} {
      background: ${theme.colors.secondary};
      border: 1px solid ${theme.colors.secondary};

      &:hover {
        border: 1px solid ${theme.colors.darkerGray};
      }
    }

    &:checked + ${Slider}:before {
      transform: translateX(1.9rem);
      background: ${theme.colors.primary};
    }

    &:hover + ${Slider} {
      border: 1px solid ${theme.colors.secondary};
    }
  `}
`;

export const Slider = styled.span`
  ${({ theme }) => css`
    position: absolute;
    cursor: pointer;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    background: ${theme.colors.darkerGray};
    border: 1px solid ${theme.colors.darkerGray};
    -webkit-transition: all 300ms ease-in-out;
    transition: all 300ms ease-in-out;
    border-radius: 2rem;

    &:before {
      position: absolute;
      content: "";
      height: 2.1rem;
      width: 2.1rem;
      left: 0.4rem;
      bottom: 0.12rem;
      background: ${theme.colors.white};
      -webkit-transition: all 300ms ease-in-out;
      transition: all 300ms ease-in-out;
      border-radius: 50%;

    }
  `}
`;
