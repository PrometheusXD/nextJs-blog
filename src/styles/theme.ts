export const theme = {
  name: 'default',
  colors: {
    primary: '#030518',
    darkText: '#333333',
    secondary: '#f231a5',
    white: '#FFFFFF',
    mediumGray: '#DDDDDD',
    darkerGray: '#AAAAAA',
  },
  font: {
    family: {
      default: "'Open Sans', sans-serif",
    },
    sizes: {
      xsmall: '8rem',
      small: '1.6rem',
      medium: '2.4rem',
      large: '3.2rem',
      xlarge: '4.0rem',
      xxlarge: '4.8rem',
      huge: '5.6rem',
      xhuge: '6.4rem',
    },
  },
  media: {
    lteMedium: '(max-width: 768px)',
    lteSmall: '(max-width: 360px)',
  },
  spacings: {
    xsmall: '0.8rem',
    small: '1.6rem',
    medium: '2.4rem',
    large: '3.2rem',
    xlarge: '4.0rem',
    xxlarge: '4.8rem',
    huge: '5.6rem',
    xhuge: '6.4rem',
  },
  sizes: {
    max: '96rem',
    content: '80rem',
  },
};
