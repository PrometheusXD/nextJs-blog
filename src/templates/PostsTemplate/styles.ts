import styled, { css } from 'styled-components';

export const ButtonContainer = styled.div`
  ${({ theme }) => css`
    padding: 0 ${theme.spacings.large};
    margin: ${theme.spacings.large} 0;
    display: flex;
    justify-content: center;
  `}
`;

export const Button = styled.button`
  ${({ theme }) => css`
    background: ${theme.colors.primary};
    color: ${theme.colors.white};
    border: none;
    padding: ${theme.spacings.small} ${theme.spacings.large};
    cursor: pointer;
    transition: all 300ms ease-in-out;


    &:disabled,
    &:disabled:hover {
      background: ${theme.colors.darkerGray};
      color: #FFFFFF;
    }

    &:hover {
      background: ${theme.colors.secondary};
      color: #FFFFFF;
    }
  `}
`;
