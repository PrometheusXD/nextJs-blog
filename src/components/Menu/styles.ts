import styled, { DefaultTheme, css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

type MenuBehaviorProps = {
  menuVisible: boolean;
};

const wrapperChanger = (theme: DefaultTheme) =>
  css`
    left: -30rem;
    overflow-y: hidden;

    @media ${theme.media.lteMedium} {
    left: -32rem;
    }
`;

export const Wrapper = styled.div<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    background: ${theme.colors.primary};
    padding: ${theme.spacings.large};
    display: flex;
    position: fixed;
    z-index: 1;
    width: 100%;
    max-width: 32rem;
    height: 100vh;
    top: 0;
    left: 0;
    transition: all 300ms ease-in-out;
    overflow-y: auto;

    ${!menuVisible && wrapperChanger(theme)}
  `}
`;

export const Nav = styled.nav`
  margin: auto;
  width: 100%;
`;

export const Logo = styled.div`
  ${({ theme }) => css`
    ${HeadingStyles} {
      display: flex;
      justify-content: center;
      margin: 0;
      margin-bottom: ${theme.spacings.xlarge};
    }
  `}
`;

const buttonChanger = (theme: DefaultTheme) =>
  css`
    left: 1rem;

    @media ${theme.media.lteMedium} {
    left: -0.5rem;
    }
`;

export const OpenClose = styled.a<MenuBehaviorProps>`
  ${({ theme, menuVisible }) => css`
    position: fixed;
    top: ${theme.spacings.medium};
    color: ${theme.colors.white};
    background: ${theme.colors.primary};
    z-index: 2;
    width: 3rem;
    height: 3rem;
    left: 26rem;
    transition: all 300ms ease-in-out;

    &:hover {
    color: ${theme.colors.secondary};
    }

    ${!menuVisible && buttonChanger(theme)}
  `}
`;
