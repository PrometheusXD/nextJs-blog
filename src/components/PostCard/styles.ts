import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    position: relative;
    box-shadow: 0 0 5px 0px ${theme.colors.darkerGray};

    &::after,
    &::before {
      content: "";
      position: absolute;
      top: 0;
      bottom: 0;
      left: 0;
      right: 0;
      transition: transform .3s ease-in-out;
    }

    &::before {
      border-left: 2px solid ${theme.colors.primary};
      border-right: 2px solid ${theme.colors.primary};
      transform: scaleY(0);
    }

    &::after {
      border-top: 2px solid ${theme.colors.primary};
      border-bottom: 2px solid ${theme.colors.primary};
      transform: scaleX(0);
    }

    &:hover::before {
      transform: scaleY(1);
    }

    &:hover::after {
      transform: scaleX(1);
    }


    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacings.small};
    }
  `}
`;

export const Cover = styled.img`
  max-width: 100%;
`;

export const Content = styled.div`
  ${({ theme }) => css`
    padding: calc(${theme.spacings.small} - 0.6rem);
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.darkText};
  `}
`;
