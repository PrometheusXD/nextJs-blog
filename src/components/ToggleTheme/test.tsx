import { renderTheme } from '../../styles/render-theme';
import { ToggleTheme } from '.';
import { theme } from '../../styles/theme';
import { fireEvent, screen } from '@testing-library/react';

describe('<ToggleTheme />', () => {
  beforeEach(() => {
    localStorage.setItem(
      'theme',
      JSON.stringify({ ...theme, name: 'inverted' }),
    );
  });
  afterEach(() => {
    localStorage.removeItem('theme');
  });

  it('should render input without theme', () => {
    localStorage.removeItem('theme');
    renderTheme(<ToggleTheme />);

    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });

  it('should render input with default theme', () => {
    localStorage.setItem('theme', JSON.stringify({ name: 'default' }));
    renderTheme(<ToggleTheme />);

    expect(screen.getByRole('checkbox')).not.toBeChecked();
  });

  it('should render input checked with inverted theme', () => {
    renderTheme(<ToggleTheme />);

    expect(screen.getByRole('checkbox')).toBeChecked();
  });

  it('should render input and toggle between theme when toggling input', () => {
    const { container } = renderTheme(<ToggleTheme />);

    const label = screen.getByLabelText('Toggle light and dark mode');
    const input = screen.getByRole('checkbox');

    fireEvent.click(label);
    expect(input).not.toBeChecked();

    fireEvent.click(label);
    expect(input).toBeChecked();

    expect(container).toMatchSnapshot();
  });
});
