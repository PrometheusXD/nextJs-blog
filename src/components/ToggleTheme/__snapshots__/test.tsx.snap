// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[
  `<ToggleTheme /> should render input and toggle between theme when toggling input 1`
] = `
.c0 {
  position: fixed;
  top: 2rem;
  right: 2rem;
  z-index: 2;
}

.c1 {
  position: relative;
  display: inline-block;
  width: 5rem;
  height: 2.5rem;
  line-height: 0;
  font-size: 0;
  overflow: hidden;
  color: rgba(0,0,0,0);
}

.c5:checked + .c3 {
  background: #f231a5;
  border: 1px solid #f231a5;
}

.c5:checked + .c3:hover {
  border: 1px solid #AAAAAA;
}

.c5:checked + .c3:before {
  -webkit-transform: translateX(1.9rem);
  -ms-transform: translateX(1.9rem);
  transform: translateX(1.9rem);
  background: #030518;
}

.c5:hover + .c3 {
  border: 1px solid #f231a5;
}

.c2 {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  opacity: 0;
  width: 0;
  height: 0;
}

.c2:checked + .c3 {
  background: #f231a5;
  border: 1px solid #f231a5;
}

.c2:checked + .c3:hover {
  border: 1px solid #CCCCCC;
}

.c2:checked + .c3:before {
  -webkit-transform: translateX(1.9rem);
  -ms-transform: translateX(1.9rem);
  transform: translateX(1.9rem);
  background: #FFFFFF;
}

.c2:hover + .c3 {
  border: 1px solid #f231a5;
}

.c4 {
  position: absolute;
  cursor: pointer;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: #CCCCCC;
  border: 1px solid #CCCCCC;
  -webkit-transition: all 300ms ease-in-out;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  border-radius: 2rem;
}

.c4:before {
  position: absolute;
  content: "";
  height: 2.1rem;
  width: 2.1rem;
  left: 0.4rem;
  bottom: 0.12rem;
  background: #0A1128;
  -webkit-transition: all 300ms ease-in-out;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  border-radius: 50%;
}

<div>
  <div
    class="c0"
  >
    <label
      class="c1"
    >
      Toggle light and dark mode
      <input
        class="c2"
        type="checkbox"
        value="Dark mode active"
      />
      <span
        class="c3 c4"
      />
    </label>
  </div>
</div>
`;
